src/modules/Tasklist.js

```diff
@@ -0,0 +1,239 @@
+      if (completed) {
+        checkbox.checked = true;
+      } else {
+        checkbox.checked = false;
+      }
+      const li = document.createElement('li');
+      li.id = `li${i}`;
+      const textarea = document.createElement('textarea');
+      textarea.innerHTML = description;
+      textarea.classList.add('description');
+      textarea.rows = 1;
+      if (completed) { textarea.classList.add('tachado'); }
+      li.appendChild(checkbox);
+      li.appendChild(textarea);
+      mainlist.appendChild(li);
+      checkbox.addEventListener('change', this.#checkchange);
+      this.#textinput2(textarea);
+      textarea.addEventListener('input', this.#textinput, false);
+      textarea.addEventListener('keypress', this.#checkenter);
+      textarea.addEventListener('change', this.#textchanged);
+      textarea.addEventListener('focus', this.#focused);
+    }
+    const divfinal = document.createElement('div');
+    divfinal.innerHTML = 'Clear All Completed';
+    divfinal.classList.add('divfinal');
+    const li = document.createElement('li');
+    li.appendChild(divfinal);
+    mainlist.appendChild(li);
+    divfinal.addEventListener('click', this.#clearcompleted);
+  };
+
+  #clearcompleted = () => {
+    const result = this.tasks.filter((task) => task.completed === false);
+    this.tasks = [...result];
+    for (let i = 0; i < this.tasks.length; i += 1) {
+      this.tasks[i].index = i + 1;
+    }
+    this.savedata();
+    this.#clearlist();
+    this.renderdata();
+  };
+
+  #textchanged = (event) => {
+    const thisid = parseInt(event.target.parentNode.id.substring(2), 10);
+    this.tasks[thisid].description = event.target.value;
+    if (event.target.value === '') {
+      this.tasks.splice(thisid, 1);
+      for (let i = thisid; i < this.tasks.length; i += 1) {
+        this.tasks[i].index -= 1;
+      }
+    }
+    this.savedata();
+    this.#clearlist();
+    this.renderdata();
+  };
+
+  #deleteitem = (event) => {
+    const thisid = parseInt(event.target.parentNode.id.substring(2), 10);
+    this.tasks.splice(thisid, 1);
+    for (let i = thisid; i < this.tasks.length; i += 1) {
+      this.tasks[i].index -= 1;
+    }
+    this.savedata();
+    this.#clearlist();
+    this.renderdata();
+  };
+
+  #blurred = (event) => {
+    const parentli = event.target.parentNode;
+    const recyclebin = parentli.querySelector('.recyclebin');
+    if (recyclebin !== null) recyclebin.remove();
+    if (parentli.classList.contains('colorbg')) parentli.classList.remove('colorbg');
+    if (event.target.classList.contains('colorbg')) event.target.classList.remove('colorbg');
+  };
+
+  #focused = (event) => {
+    const parentli = event.target.parentNode;
+    if (parentli.querySelector('.recyclebin') === null) {
+      const recyclebin = document.createElement('img');
+      recyclebin.classList.add('recyclebin');
+      recyclebin.src = recyclebinimg;
+      recyclebin.height = '15';
+      recyclebin.width = '15';
```

Comment:
Since you are adding a class to the recycle bin image, why not include in the same class height and width of the image element?

Reply:
Good question, I want the old HTML attributes to be set because that reserves the 15x15 space for the image, while it's loading. I'm not sure if setting in CSS the width and height does the same, it probably does. I'll give it a try.

Reply:
It seems to be tidier to put them in the CSS, no benefits from putting them as HTML attributes.